### ISSUE TITLE
Background color for selected row when tree is edited can be inconsistent

### ISSUE THREAD

**Issue by ingokegel** [JFormDesigner/FlatLaf#445]
The current behavior is that In `FlatTreeUi.paintRow`, when the tree is in editing mode, the selected row will be painted with an inactive background.

However, some editors are only active during the duration of a click, such as in a tree with checkboxes. In that case, the item listener of the checkbox will typically stop editing. During the duration of the click on a selected row, the check box is drawn with the inactive background, This will not affect the whole row because of the clip bounds.

![2021-12-16_16-36-20](https://user-images.githubusercontent.com/1199475/146401695-522528a9-22e9-420f-a5dc-9d469ab0fda7.png)

I currently work around this problem with a custom FlatTreeUI:

```
    @Override
    public void updateUI() {
        setUI(new FlatTreeUI() {
            @Override
            protected void paintRow(Graphics g, Rectangle clipBounds, Insets insets, Rectangle bounds, TreePath path, int row, boolean isExpanded, boolean hasBeenExpanded, boolean isLeaf) {
                boolean isEditing = (editingComponent != null && editingRow == row);
                boolean isSelected = tree.isRowSelected(row);
                if (isEditing) {
                    if (isSelected && isWideSelection()) {
                        Color oldColor = g.getColor();
                        g.setColor(selectionBackground);
                        paintWideSelection(g, clipBounds, insets, bounds, path, row, isExpanded, hasBeenExpanded, isLeaf);
                        g.setColor(oldColor);
                    }
                    return;
                }

                super.paintRow(g, clipBounds, insets, bounds, path, row, isExpanded, hasBeenExpanded, isLeaf);
            }

            private void paintWideSelection(Graphics g, Rectangle clipBounds, Insets insets, Rectangle bounds, TreePath path, int row, boolean isExpanded, boolean hasBeenExpanded, boolean isLeaf) {
                g.fillRect(0, bounds.y, tree.getWidth(), bounds.height);
                if (shouldPaintExpandControl(path, row, isExpanded, hasBeenExpanded, isLeaf)) {
                    paintExpandControl(g, clipBounds, insets, bounds, path, row, isExpanded, hasBeenExpanded, isLeaf);
                }
            }
        });
    }
```

where the only difference to the super implementation is `g.setColor(selectionBackground);` instead of `g.setColor(selectionInactiveBackground);`.

Is there a possibility to make this behavior configurable and if yes, what would be the best strategy? I'm of course willing to provide a pull request and test it on my side.

**Comment by DevCharly** [comment]
> Is there a possibility to make this behavior configurable and if yes, what would be the best strategy? 

Good question. Have no idea at the moment.
That's a very special case. Never seen a checkbox as tree cell editor...

Maybe it is best to avoid going into editing mode?
Can't you simple toggle the checkbox on mouse pressed (or released)?
JIDE `com.jidesoft.swing.CheckBoxTree` seems to work fine that way.

**Comment by ingokegel** [comment]
You are right, not going into editing mode is much better and prevents other problems as well for which I had to add workarounds. When I first implemented my check tree way before jide existed, I based it on samples like this one:

http://www.java2s.com/Code/Java/Swing-JFC/CheckBoxNodeTreeSample.htm

So I guess this is not too uncommon. In any case, I have migrated my check tree to not use an editor and this is not an issue for me anymore.
